hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -735,15 +768,81 @@ public void run() {
                     LOGGER.info("Exception while watching for StatefulSet changes", e);
                     try {
                         watchResponse.disconnect();
-                    } catch (Throwable t) {
+                    } catch (Exception t) {
                         LOGGER.fine("Exception while closing connection after an IOException", t);
                     }
                 }
             }
         }
 
+        private void handleFailure(BackoffIdleStrategy backoffIdleStrategy, RestClientException e) {
+            if (e.getHttpErrorCode() == HTTP_GONE) {
+                // occurs when the resource version we are watching for is stale
+                LOGGER.info("Statefulset watcher has fallen behind, re-reading sts list and resuming watch: "
+                        + e.getMessage());
+            } else {
+                // watch failed with another HTTP error code, let's log at WARNING level,
+                // backoff and try to resume again
+                LOGGER.warning("Error while attempting to watch kubernetes API for statefulsets: "
+                        + e.getHttpErrorCode() + " " + e.getMessage() + ". Backing off (n: " + idleCount
+                        + " ) before ");
```

Comment:
```suggestion
                        + ")");
```
It seems this log cut in half.

Reply:
yup, fixed in 35f52e7d2e5a57e6062dd409e88c557daf676752